src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -672,767 +486,438 @@ private void ProcessWSManProtocolForTestConnection()
+    [SuppressMessage("Microsoft.Design", "CA1027:MarkEnumsWithFlags")]
+    public enum WaitForServiceTypes
     {
-#region Parameters
         /// <summary>
-        /// Specifies the Drive on which the system restore will be enabled.
-        /// The drive string should be of the form "C:\".
+        /// Wait for the WMI service to be ready
         /// </summary>
-        [Parameter(Position = 0, Mandatory = true)]
-        [ValidateNotNullOrEmpty]
-        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
-        public string[] Drive { get; set; }
+        Wmi = 0x0,
+
+        /// <summary>
+        /// Wait for the WinRM service to be ready
+        /// </summary>
+        WinRM = 0x1,
 
-        #endregion Parameters
+        /// <summary>
+        /// Wait for the PowerShell to be ready
+        /// </summary>
+        PowerShell = 0x2,
+    }
 
-        private ManagementClass WMIClass;
-        private const string ErrorBase = "ComputerResources";
-#region "IDisposable Members"
+    /// <summary>
+    /// Restarts the computer
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Restart, "Computer", SupportsShouldProcess = true, DefaultParameterSetName = DefaultParameterSet,
+        HelpUri = "https://go.microsoft.com/fwlink/?LinkID=135253", RemotingCapability = RemotingCapability.OwnedByCommand)]
+    public class RestartComputerCommand : PSCmdlet, IDisposable
+    {
+#region "Parameters and PrivateData"
+
+        private const string DefaultParameterSet = "DefaultSet";
+        private const string AsJobParameterSet = "AsJobSet";
 
         /// <summary>
-        /// Dispose Method
+        /// Used to start a command remotely as a Job. The Job results are collected
+        /// and stored in the global cache on the client machine.
         /// </summary>
-        public void Dispose()
-        {
-            this.Dispose(true);
-            // Use SuppressFinalize in case a subclass
-            // of this type implements a finalizer.
-            GC.SuppressFinalize(this);
-        }
+        [Parameter(ParameterSetName = AsJobParameterSet)]
+        public SwitchParameter AsJob { get; set; } = false;
 
         /// <summary>
-        /// Dispose Method.
+        /// The following is the definition of the input parameter "Authentication".
+        /// Specifies the authentication level to be used with WMI connection. Valid
+        /// values are:
+        ///
+        /// Unchanged = -1,
+        /// Default = 0,
+        /// None = 1,
+        /// Connect = 2,
+        /// Call = 3,
+        /// Packet = 4,
+        /// PacketIntegrity = 5,
+        /// PacketPrivacy = 6.
         /// </summary>
-        /// <param name="disposing"></param>
-        public void Dispose(bool disposing)
+        [Parameter]
+        [Alias("Authentication")]
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly")]
+        public AuthenticationLevel DcomAuthentication
```

Comment:
We should remove this since DcomAuthentication isn't supported.  Will need documentation update.